canonical-web-and-design/charmhub.io
# Topic pages backend integration

## Done
- Removed previous hardcoded Kubernetes topic 
- Topic pages integration with Discourse: `topics/<string:topic_slug>`
- Used `exclude_xframe_options_header` decorator for embeddable cards
- Added list of topics on `webapp/topics/topics.json`
- Added topics.json endpoint: `/topics.json` with query parameter `/topics.json?q=foo`

## QA
Check the content on the individual topic pages:
- https://charmhub-io-1231.demos.haus/topics/kubernetes
- https://charmhub-io-1231.demos.haus/topics/kubeflow
- https://charmhub-io-1231.demos.haus/topics/openstack

Check the JSON endpoint:
- https://charmhub-io-1231.demos.haus/topics.json should return 3 items
- https://charmhub-io-1231.demos.haus/topics.json?q=kube should return 2 items
- https://charmhub-io-1231.demos.haus/topics.json?q=featured should return 3 items
- https://charmhub-io-1231.demos.haus/topics.json?q=nothing should return 0 items
